assets/quick-add.js

```diff
@@ -0,0 +1,83 @@
+if (!customElements.get('quick-add-modal')) {
+  customElements.define('quick-add-modal', class QuickAddModal extends ModalDialog {
+    constructor() {
+      super();
+      this.modalContent = this.querySelector('[id^="QuickAddInfo-"]');
+    }
+
+    hide(preventFocus = false) {
+      const cartNotification = document.querySelector('cart-notification');
+      if (cartNotification) cartNotification.setActiveElement(this.openedBy);
+      this.modalContent.innerHTML = '';
+      
+      if (preventFocus) this.openedBy = null;
+      super.hide();
+    }
+
+    show(opener) {
+      opener.setAttribute('aria-disabled', true);
+      opener.classList.add('loading');
+      opener.querySelector('.loading-overlay__spinner').classList.remove('hidden');
+
+      fetch(opener.getAttribute('data-product-url'))
+        .then((response) => response.text())
+        .then((responseText) => {
+          const responseHTML = new DOMParser().parseFromString(responseText, 'text/html');
+          const productInfo = responseHTML.querySelector('section[id^="MainProduct-"]');
+
+          this.setInnerHTML(this.modalContent, this.removeDOMElements(productInfo).innerHTML);
+          
+          if (window.Shopify && Shopify.PaymentButton) {
+            Shopify.PaymentButton.init();
+          }
+
+          if (window.ProductModel) window.ProductModel.loadShopifyXR();
+
+          this.removeGalleryListSemantic();
+          this.preventVariantURLSwitching();
+          super.show(opener);
+        })
+        .finally(() => {
+          opener.removeAttribute('aria-disabled');
+          opener.classList.remove('loading');
+          opener.querySelector('.loading-overlay__spinner').classList.add('hidden');
+        });
+    }
+
+    setInnerHTML(element, html) {
+      element.innerHTML = html;
+
+      // Reinjects the script tags to allow execution. By default, scripts are disabled when using element.innerHTML.
+      element.querySelectorAll('script').forEach(oldScriptTag => {
+        const newScriptTag = document.createElement('script');
+        Array.from(oldScriptTag.attributes).forEach(attribute => {
+          newScriptTag.setAttribute(attribute.name, attribute.value)
+        });
+        newScriptTag.appendChild(document.createTextNode(oldScriptTag.innerHTML));
+        oldScriptTag.parentNode.replaceChild(newScriptTag, oldScriptTag);
+      });
+    }
+
+    preventVariantURLSwitching() {
+      this.modalContent.querySelector('variant-radios,variant-selects').setAttribute('data-update-url', 'false');
+    }
```

Comment:
There is something happening when switching variant that are linked to images. It's related to the modal not cutting the content outside of it's limit/size. 
[video](https://screenshot.click/21-49-iqz5v-3377z.mp4)